### PR TITLE
Add CCZ4 functions for the gradient of the gradient of the lapse and the divergence of the lapse

### PR DESCRIPTION
**Pull request by macedo22**
## Proposed changes

As part of implementing the first order CCZ4 system, this PR adds functionx for computing:
- the gradient of the gradient of the lapse
- the divergence of the lapse

The motivation for this is to compute eq. (21) and (22) in this [CCZ4 paper](https://arxiv.org/pdf/1707.09910.pdf), whose values will later be necessary for computing the evolved variables.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Comment by macedo22** [comment]
@nilsdeppe Compute tags for these functions have been removed and it is now ready for review

File: src/Evolution/Systems/Ccz4/DerivLapse.hpp
Comment (nilsdeppe) [inline src/Evolution/Systems/Ccz4/DerivLapse.hpp]: `grad_grad_lapse_` -> `result` or `result_grad_grad_lapse`? We use the `_` suffix to denote private member variables

File: src/Evolution/Systems/Ccz4/DerivLapse.cpp
Comment (nilsdeppe) [inline src/Evolution/Systems/Ccz4/DerivLapse.cpp]: isn't this for loop:
```cpp
get(div_lapse) = 0.0;
```
?

File: src/Evolution/Systems/Ccz4/DerivLapse.cpp
Comment (nilsdeppe) [inline src/Evolution/Systems/Ccz4/DerivLapse.cpp]: `conformal_factor.get()` -> `get(conformal_factor)`

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
Overall looks good, couple minor things you can squash immediately

File: src/Evolution/Systems/Ccz4/DerivLapse.cpp
Comment (macedo22) [reply src/Evolution/Systems/Ccz4/DerivLapse.cpp]: Yes, I recycled this for loop from another function and didn't think about how it's not needed here

**Comment by macedo22** [comment]
@nilsdeppe Updated!

**Review by nilsdeppe — DISMISSED** [review]

**Comment by macedo22** [comment]
@nilsdeppe Sorry to have dismissed your previous approval. Per your request on my other PR [here](https://github.com/sxs-collaboration/spectre/pull/3517#discussion_r710537178), I've just fixed the notation in this PR to be consistent by changing `\bar` -> `\tilde`, so it will need to be reapproved.

Edit: I've also fixed a couple other change requests from your review of that PR but here:
- Removed the `Tensor.hpp` include from `DerivLapse.hpp`
- Moved the include for `DerivLapse.hpp` to be above system includes
- Removed `TypeAliases.hpp` and `IndexType.hpp` includes from `DerivLapse.cpp`

File: src/Evolution/Systems/Ccz4/DerivLapse.hpp
Comment (nilsdeppe) [inline src/Evolution/Systems/Ccz4/DerivLapse.hpp]: You should have the `tilde` over the indices. That is `\tilde{\gamma}^{ij}` not `\tilde{\gamma^{ij}}`

**Review by nilsdeppe — COMMENTED** [review]
You can squash immediately

**Comment by macedo22** [comment]
@nilsdeppe Updated and rebased!

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**
